File: java/app/src/main/java/githubtododemo/githubclient/models/AuthorAssociation.java
package githubtododemo.githubclient.models;

import com.microsoft.kiota.serialization.ValuedEnum;
import java.util.Objects;

/** How the author is associated with the repository. */
public enum AuthorAssociation implements ValuedEnum {
    COLLABORATOR("COLLABORATOR"),
    CONTRIBUTOR("CONTRIBUTOR"),
    FIRST_TIMER("FIRST_TIMER"),
    FIRST_TIME_CONTRIBUTOR("FIRST_TIME_CONTRIBUTOR"),
    MANNEQUIN("MANNEQUIN"),
    MEMBER("MEMBER"),
    NONE("NONE"),
    OWNER("OWNER");
    public final String value;
    AuthorAssociation(final String value) {
        this.value = value;
    }
    @javax.annotation.Nonnull
    public String getValue() { return this.value; }
    @javax.annotation.Nullable
    public static AuthorAssociation forValue(@javax.annotation.Nonnull final String searchValue) {
        Objects.requireNonNull(searchValue);
        switch(searchValue) {
            case "COLLABORATOR": return COLLABORATOR;
            case "CONTRIBUTOR": return CONTRIBUTOR;
            case "FIRST_TIMER": return FIRST_TIMER;
            case "FIRST_TIME_CONTRIBUTOR": return FIRST_TIME_CONTRIBUTOR;
            case "MANNEQUIN": return MANNEQUIN;
            case "MEMBER": return MEMBER;
            case "NONE": return NONE;
            case "OWNER": return OWNER;
            default: return null;
        }
    }
}

File: php/GitHub/Models/AuthorAssociation.php
<?php

namespace Baywet\Githubtododemo\Github\Models;

use Microsoft\Kiota\Abstractions\Enum;

class AuthorAssociation extends Enum {
    public const C_O_L_L_A_B_O_R_A_T_O_R = 'COLLABORATOR';
    public const C_O_N_T_R_I_B_U_T_O_R = 'CONTRIBUTOR';
    public const F_I_R_S_T__T_I_M_E_R = 'FIRST_TIMER';
    public const F_I_R_S_T__T_I_M_E__C_O_N_T_R_I_B_U_T_O_R = 'FIRST_TIME_CONTRIBUTOR';
    public const M_A_N_N_E_Q_U_I_N = 'MANNEQUIN';
    public const M_E_M_B_E_R = 'MEMBER';
    public const N_O_N_E = 'NONE';
    public const O_W_N_E_R = 'OWNER';
}
